### ISSUE TITLE
[Google3 Bug]: MenuItemButton voice over/semantics doesn't pronounce disabled buttons

### ISSUE THREAD

**Issue by rezgoogle** [flutter/flutter#171231]
b/426553069

### Help us understand the severity of this issue

- [ ] causing severe production issues e.g. malfunctions or data loss
- [ ] blocking next binary release
- [x] blocking a client feature launch within a quarter
- [ ] nice-to-have but does not block a launch within the next quarter

### Steps to reproduce

Google Earth uses MenuItemButton which calls private member _MenuItemLabel, however when the menu item is disabled the voice over semantics doesn’t announce that it’s disabled.
https://source.corp.google.com/piper///depot/google3/third_party/dart/flutter/lib/src/material/menu_anchor.dart;l=2766

what I see happening is if 
a) there’s no semantic label provided, on line 2875, it labels it using _LocalizedShortcutLabel which doesn’t take in account the enabled state, or
b) on line 2889 it creates a Semantics(…) object if we provide a label, but again doesn’t have “enabled” field used.

this is where we use MenuItemButton: https://source.corp.google.com/piper///depot/google3/geo/earth/app/flutter/components/menu_items/lib/menu_items.dart;l=111

### Expected results

voice over pronounces that the menu item is disabled when that menu item is disabled

### Actual results

voice over calls a disabled menu item button as its regular semantic name

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I used [this](https://api.flutter.dev/flutter/material/MenuItemButton-class.html#material.MenuBar.1) code sample which has enabled and disabled menu entries and verified by running on iOS voiceover. 
The enabled entry is read as: `Show Message control S`. 
The disabled entry is `Reset Message Esc` which is read as: `Reset message esc dim`.

Based on the findings as mentioned in issue description, I'll keep the issue open and label for team's tracking.

```
stable : 3.32.5
master : 3.33.0-1.0.pre.677
```
